netplan/cli/commands/ip.py

```diff
@@ -147,6 +147,9 @@ def lease_method_nm_connection():
             out = subprocess.check_output(argv, universal_newlines=True)
         except CalledProcessError:  # pragma: nocover (better be covered in autopkgtest)
             sys.exit(1)
+        if not out:
```

Comment:
So... I really wonder how `out` can be `None` here?

We're calling `subprocess.check_output()` above, which should throw a "CalledProcessError" in case `generate --mapping ...` returns non-zero, which it should in case of missing ID. Do you know what's the root cause for this? Do we need to fix `generate` to actually return non-zero in all error cases?